demo/src/lib/gtagEvents.js

```diff
@@ -0,0 +1,132 @@
+import {v4} from 'uuid';
+import {event} from './gtag.js';
+import {getItemParameters, getItemsArrayFromCart, computePriceOfItemsInCart} from '../utils.js';
+
+/**
+ * Sends a select_item event to Google Analytics.
+ * View the [Ecommerce Guide](https://developers.google.com/analytics/devguides/collection/app-web/ecommerce)
+ * to review the event and its accepted parameters.
+ * @param {string} itemId
+ */
+export function sendSelectItemEvent(itemId) {
+  event('select_item', {
+    items: [{
+      ...getItemParameters(itemId),
+    }],
+  });
+}
+
+/**
+ * Sends a view_item event to Google Analytics.
+ * View the [Ecommerce Guide](https://developers.google.com/analytics/devguides/collection/app-web/ecommerce)
+ * to review the event and its accepted parameters.
+ * @param {string} itemId
+ */
+export function sendViewItemEvent(itemId) {
+  event('view_item', {
+    items: [{
+      ...getItemParameters(itemId),
+    }],
+  });
+}
+
+/**
+ * Sends an add_to_cart event to Google Analytics.
+ * View the [Ecommerce Guide](https://developers.google.com/analytics/devguides/collection/app-web/ecommerce)
+ * to review the event and its accepted parameters.
+ * @param {string} itemId
+ */
+export function sendAddToCartEvent(itemId) {
+  event('add_to_cart', {
+    items: [{
+      ...getItemParameters(itemId),
+    }],
+  });
+}
+
+/**
+ * Sends a remove_from_cart event to Google Analytics.
+ * View the [Ecommerce Guide](https://developers.google.com/analytics/devguides/collection/app-web/ecommerce)
+ * to review the event and its accepted parameters.
+ * @param {string} itemId
+ */
+export function sendRemoveFromCartEvent(itemId) {
+  event('remove_from_cart', {
+    items: [{
+      ...getItemParameters(itemId),
+    }],
+  });
+}
+
+/**
+ * Sends a view_cart event to Google Analytics.
+ * View the [Ecommerce Guide](https://developers.google.com/analytics/devguides/collection/app-web/ecommerce)
+ * to review the event and its accepted parameters.
+ */
+export function sendViewCartEvent() {
+  event('view_cart', {
+    value: computePriceOfItemsInCart(),
+    currency: 'USD',
+    items: getItemsArrayFromCart(),
+  });
+}
```

Comment:
nit: This pattern is also repeated (and extended below). Any chance you can think of a way to dry this up?

Not necessarily recommending you do this, just want to get your thoughts.